### DISCUSSION TITLE
How to convert with  a formula in groovy from #.###,00 to ###.00?

### DISCUSSION THREAD

**Discussion by Illegit** [freeplane/freeplane#769] [Help] [answered]
How to convert with a formula in groovy from #.###,00 to ###.00?
if I have the value 13.000,34  because I need to use in FP the value 13000.34, which formula I need to use?
Thank you

**Comment by euu2021** [comment, marked as answer]
I'm not sure if I understand your question, but I suceeded in this experiment, that may be useful to you:

I have a node with the text 13.000,34 and I apply a script that formats it to be 1300.34 (see the status line):

![image](https://user-images.githubusercontent.com/77707706/197407670-e2dd7fe7-7b7d-4192-9c4f-91c7d787db67.png)

This is the script:
```
import java.text.NumberFormat;
import java.util.Locale;

c.statusInfo = NumberFormat.getNumberInstance(Locale.GERMANY).parse(node.text)
```
So, now I can take that string and use it as a number, like summing 3 to it (result in the status line):

![image](https://user-images.githubusercontent.com/77707706/197407915-4276c839-7a42-4abd-ad73-aa28f5063e0a.png)

And the script for this last one:


```
import java.text.NumberFormat;
import java.util.Locale;

n = NumberFormat.getNumberInstance(Locale.GERMANY).parse(node.text)

c.statusInfo = (n + 3)
```

**Reply by macmarrum** [reply]
Somebody's been learning Java :wink: 
Cool, man :+1:

**Comment by macmarrum** [comment]
@Illegit 
I would use the following formula (assuming your node has ID_12345)
```groovy
=ID_12345.text.replace('.':'', ',':'.') as BigDecimal
```

To convert the number-as-text into a number, I would write a script
```groovy
node.text = node.text.replace('.':'', ',':'.') as BigDecimal
```
And just to make sure the script only changes nodes which are considered text, I'd add a condition
```groovy
if (node.to.class.simpleName == 'ConvertibleNodeText')
    node.text = node.text.replace('.':'', ',':'.') as BigDecimal
```

The formula/script uses the Groovy magic, which makes it simpler than in Java.

**Comment by Illegit** [comment]
Hi, I thank euu2021,  answer working perfectly.
And I thank macmarrum, I really appreciated your script, and I think I'll use always that.
thank you I got plenty of numbers in that format to calc in FP.
